packages/botonic-dx/sample-config/package.json

```diff
@@ -16,8 +17,19 @@
 
     "########## PUBLISH": "",
     "prepublishOnly": "npm run build && npm run lint-ci && npm test",
-    "preversion": "npm run lint",
-    "version": "git add -A src tests",
+    "preversion": "npm run lint-ci",
+    "# version issue": "See https://github.com/npm/cli/issues/2010. Temporarily checking all files have been git-added",
+    "version": "git add package.json && git diff --exit-code && git commit",
     "postversion": "git push && git push --tags"
```

Comment:
As I commented [here](https://github.com/hubtype/botonic/pull/1465#discussion_r638544484), why not adding the `package-lock.json` in the `git add` command? If you don't do it, it will remain changed after the commit is done.

Also, added all the commands in the `postversion` script because doing it in the `version` script wasn't adding the files correctly.

This is my suggestion:
```
"preversion": "npm run lint-ci",
"postversion": "git add package.json package-lock.json && git diff --exit-code && git commit && git push && git push --tags"
```

Reply:
My idea was not even adding package-lock.json to git because:
* these packages only incorporate dev dependencies. So, it's not critical to freeze to a specific version because they won't reach production
* lock file is long and add noise when visually inspecting diffs between commits
* when there are merge conflicts, it's a pain to merge lock files 

@marcpdw do you think we should add package-lock.json to git for DX packages?

Reply:
do we still need to move the commands to postversion if we remove package-lock.json?

Reply:
If the idea is to not add `package-lock.json` file to git, we should add it to `.gitignore` right? I don't see it there, am I missing something?. If we don't git ignore the file, it will be modified anyway when doing `npm version` and it will be uncommited with the actual scripts configuration.

If we ignore `package-lock.json` in git, the scripts are fine as they are now, but if we want to include `package-lock.json`, we have to do `git add` of both files in the `postversion` script because otherwise the commit will include only the `package.json` file.

Reply:
thanks, good point, I'll add package-lock.json to .gitignore if @marcrabat & @marcpdw agree on my [rationale](https://github.com/hubtype/botonic/pull/1570#discussion_r644832062) for not adding the lock file to git

Reply:
it sounds reasonable to me @dpinol . Beware adding package-lock in root's gitignore, as it would ignore it for all packages. I understand that adding inside this folder would override the one in the root, is it like this?

Reply:
My opinion is that we either add `package-lock` or we set the major versions of those packages with the caret character `^` in front.
We should have it fixed for the compatibility of the plugins. I tried to use the same config as _@botonic/dx_ on `@hubtype/eslint-config` and one of the plugins was already deprecated with the version of _eslint_ I installed there.

Reply:
@asastre I added package-lock.json to .gitignore in dx & eslint packages.
@marcpdw we user ^ for most dependencies, except:
 * 2 dependencies didn't have it . fixed  in #1602
 * I don't use it for @botonic/eslint-config because, since we're still in pre v1 version, we break sometimes break compatibility when upgrading the minor cunk of the version